Tools/Scripts/libraries/resultsdbpy/resultsdbpy/__init__.py

```diff
@@ -67,7 +67,7 @@ def _maybe_add_webkit_python_library_paths():
 AutoInstall.register(Package('hiredis', Version(1, 1, 0)))
 AutoInstall.register(Package('isodate', Version(0, 6, 0)))
 AutoInstall.register(Package('jmespath', Version(0, 10, 0), wheel=True))
-AutoInstall.register(Package('lupa', Version(1, 13)))
+AutoInstall.register(Package('lupa', Version(2, 0)))
```

Comment:
I think this is only going to affect Apple-hosted servers (like `results.webkit.org`), which makes me a bit nervous, especially given they aren't actually going to be moving to Python 3.12 any time soon.

Reply:
Hm, honestly I have no clue what any of this is for. :) Basically all I'm doing here is run `test-webkitpy`, then if it doesn't crash and if EWS is happy, I assume it's good.

I could guard this behind a python 3.12 version check if desired. Or we could give it a try and add the version check only if we discover that it breaks something. Whichever you prefer.

Reply:
BTW the problem here was it failed to compile:

```
Next problem occurs when compiling lupa. There are a bunch of GCC warnings and then many errors. The first error is:

lupa/_lupa.c: In function ‘__Pyx_PyInt_AddObjC’:
lupa/_lupa.c:34443:51: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
34443 |         const digit* digits = ((PyLongObject*)op1)->ob_digit;
      |                                                   ^~
```

I guess the definition of `PyLongObject` must have changed at some point (python 3.12?).

Reply:
https://github.com/python/cpython/commit/c1b1f51cd1632f0b77dacd43092fb44ed5e053a9 is the change (3.12 indeed).

See https://github.com/cython/cython/issues/5238 for Cython being broken by this.